apps/dav/lib/DAV/CustomPropertiesBackend.php

```diff
@@ -376,6 +394,62 @@ private function cacheDirectory(string $path, Directory $node): void {
 		$this->userCache = array_merge($this->userCache, $propsByPath);
 	}
 
+	private function cacheCalendar(string $path, CalendarHome $node): void {
+		$calendars = $node->getChildren();
+
+		$users = [];
+		foreach ($calendars as $calendar) {
+			if ($calendar instanceof Calendar) {
+				$user = str_replace('principals/users/', '', $calendar->getPrincipalURI());
+				if (!isset($users[$user])) {
+					$users[$user] = ['calendars/' . $user];
+				}
+				$users[$user][] = 'calendars/' . $user . '/' . $calendar->getUri();
+			} elseif ( $calendar instanceof Inbox || $calendar instanceof Outbox || $calendar instanceof TrashbinHome || $calendar instanceof ExternalCalendar) {
+				$user = str_replace('principals/users/', '', $calendar->getOwner());
+				if (!isset($users[$user])) {
+					$users[$user] = ['calendars/' . $user];
+				}
+				$users[$user][] = 'calendars/' . $user . '/' . $calendar->getName();
+			}
+		}
+
+		$propertyMapper = \OCP\Server::get(PropertyMapper::class);
+
+		// user properties
+		$properties = $propertyMapper->findPropertiesByPathsAndUsers($users);
+
+		$propsByPath = [];
+		foreach ($users as $paths) {
+			foreach ($paths as $path) {
+				$propsByPath[$path] = [];
+			}
+		}
+
+		foreach ($properties as $property) {
+			$propsByPath[$property->getPropertypath()][$property->getPropertyname()] = $this->decodeValueFromDatabase($property->getPropertyvalue(), $property->getValuetype());
+		}
+		$this->userCache = array_merge($this->userCache, $propsByPath);
+		unset($propsByPath);
+
+		// published properties
+		$paths = [];
+		foreach ($users as $nestedPaths) {
+			$paths = array_merge($paths, $nestedPaths);
+		}
+		$paths = array_unique($paths);
+
+		$propsByPath = [];
+		foreach ($paths as $path) {
+			$propsByPath[$path] = [];
+		}
```

Comment:
This could be shortened to:

```suggestion
		$propsByPath = array_fill_keys(array_values($paths), []);
```